geomap/src/components/LeafletMap.vue

```diff
@@ -56,24 +59,67 @@ import {
 import MarkerCluster from "./MarkerCluster.vue";
 import { Vue } from "vue-class-component";
 import { useStore } from "vuex";
+import L from "leaflet";
+import { Alarm, Node } from "@/types";
 import { Coordinates } from "@/types";
 
 let leafletReady = ref(false);
 let leafletObject = ref("");
 let visible = ref(false);
 let map: any = ref();
 const store = useStore();
+let iconPath;
 
 let center = computed(() => {
-  const coordinates: Coordinates = store.getters['mapModule/getMapCenter']
+  const coordinates: Coordinates = store.getters["mapModule/getMapCenter"];
   return [coordinates.latitude, coordinates.longitude];
-})
+});
 
 let zoom = ref(2);
 
 let interestedNodes = computed(() => {
-  return store.getters['mapModule/getInterestedNodes'];
-})
+  return store.getters["mapModule/getInterestedNodes"];
+});
+
+/******Set marker color based on alarm severity******/
+
+let interestedAlarms = computed(() => {
+  return store.getters["mapModule/getAlarmsFromSelectedNodes"];
+});
+
+
+const setIcon = (node: Node): L.Icon | void =>  {
+  for (const alarm of interestedAlarms.value) {
+    if (node.label === alarm.nodeLabel) {
+      return L.icon({
+        iconUrl: setMarkerColor(alarm.severity),
+        shadowUrl: "https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png",
```

Comment:
We should download this file and include it in the source if we need to reference it. Some OpenNMS users are behind firewalls that don't allow external access.

Reply:
Downloaded and  replace below URL with the file path.
ShadowUrl: "https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png",